sanic/server.py

```diff
@@ -955,14 +955,14 @@ def serve(
 def _build_protocol_kwargs(
     protocol: Type[HttpProtocol], config: Config
 ) -> dict:
-    if hasattr(protocol, "websocket_timeout"):
+    if (dir(protocol).__contains__("websocket_handshake")):
```

Comment:
Why are you checking for properties this way? Please avoid using `dir()` here.

Reply:
reverted to `hasattr`, but I might circle back on this to see why we are not using `isinstance` here, which seems more logical. Do we need duck typing for some reason I am unaware of?